### PR TITLE
DEMO- 390 (feature): Custom legend

### PR DESCRIPTION
**Pull request by Paterma**
**JIRA Ticket [Custom Legend](https://rocketcom.atlassian.net/browse/DEMO-390**

## Brief Description
- Build custom legend - colors for checkboxes are not assigned yet since data will be changing

## Issues and/or Limitations
- Legend had to be custom built due to inability/complexity to have access to functions on checkboxes. With this- absolute positioning was the only way to get the legend 'inside' the chart canvas. This makes responsiveness tricky- but it is responsive to the drawers opening/closing. 

## Final Checklist

- [x] Works in Chrome
- [x] Works in Firefox
- [x] Works in Safari
- [ ] Handles responsiveness as required
- [x] Dark theme styles are correct
- [x] Light theme styles are correct

### CODE REVIEW

File: src/app/main/utility-components/track-data-utility/track-data/track-data.component.ts [hunk 1]
```diff
@@ -187,6 +199,43 @@ export class TrackDataComponent {
...
+      const index = this.selectedFilters.indexOf(filter);
+      if (index !== -1) {
+        this.selectedFilters.splice(index, 1);
+      }
+    }
+    this.filteredLegendData = this.dummyFileSize.filter((size) => {
+      return (
+        this.selectedFilters.length === 0 ||
+        this.selectedFilters.some((filter) => {
+          if (filter === 'Az') {
```
Comment (Paterma) [inline src/app/main/utility-components/track-data-utility/track-data/track-data.component.ts:219]: This is temporary filtering logic that is set up now for the new data.

**Review by NicDominguez — CHANGES_REQUESTED** [review]
Nicely done figuring out how to build a custom component over the chart. Are you able to move it down a bit? 
image

**Comment by Paterma** [comment]
> Nicely done figuring out how to build a custom component over the chart. Are you able to move it down a bit? image

I can- but if I move it down and you shrink the screen, it's outside of the chart. Take a quick look and let me know what you think. I'm happy to do it though!

**Review by NicDominguez — APPROVED** [review]
We'll probably have to revisit the responsiveness after design takes a look but this is great for now. Great job hooking up the filter logic.

**Pull request merged**